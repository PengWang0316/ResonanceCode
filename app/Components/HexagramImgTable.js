import React from "react";
import PropTypes from "prop-types";
import HexagramImage from "./HexagramImage";
// import { transferHexagramStringToImageString } from "../apis/Util";

const HexagramImgTable = (props) => {
  // console.log("props",props);
  let showArray = [];

  props.hexagramsArray.map((element)=>{
    // console.log("wilhelm_huang_hintley_name:",element.wilhelm_huang_hintley_name);
    let imgageName = element.wilhelm_huang_hintley_name ? element.wilhelm_huang_hintley_name.match(/\s*(\w+)\s*\//)[1] : "";
    showArray.push(
      <div key={element._id} className="inlineBlock hexagram-container text-center" onClick={()=>{props.onCallback(element.img_arr);}}>
        <div>
          <HexagramImage imageNumber={element.img_arr} isBlack={true}/>
        </div>
        <div className="text-center"># {element.number}</div>
        <div className="text-center">{imgageName}</div>
      </div>
    );
  });
  return (
    <div>
      {showArray}
    </div>
  );
};
HexagramImgTable.proptypes = {
  hexagramsArray: PropTypes.array.isrequired,
  onCallback: PropTypes.func.isrequired
};
export default HexagramImgTable;
