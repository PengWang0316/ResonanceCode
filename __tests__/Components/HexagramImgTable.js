import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import HexagramImgTable from '../../app/Components/HexagramImgTable';

jest.mock('../../app/Components/HexagramImage', () => 'HexagramImage');

describe('HexagramImgTable test', () => {
  const defaultProps = {
    hexagramsArray: [{
      wilhelm_huang_hintley_name: 'The Creative / Initiating / Heaven',
      img_arr: '1,2,3,4,5,6',
      _id: 'id',
      number: 'number'
    }],
    onCallback: jest.fn()
  };
  // const getShallowComponent = (props = defaultProps) => {};

  test('HexagramImgTable snapshot', () => expect(renderer.create(<HexagramImgTable {...defaultProps} />).toJSON()).toMatchSnapshot());
});
