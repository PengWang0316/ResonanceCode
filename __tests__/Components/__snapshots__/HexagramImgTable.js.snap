// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HexagramImgTable test HexagramImgTable snapshot 1`] = `
<div>
  <div
    className="d-inline-block text-center hexagramContainer"
    onClick={[Function]}
    role="button"
    tabIndex="-1"
  >
    <div>
      <HexagramImage
        imageNumber="1,2,3,4,5,6"
        isBlack={true}
        isFirstImage={true}
      />
    </div>
    <div
      className="text-center"
    >
      # 
      number
    </div>
    <div
      className="text-center"
    >
      The Creative
    </div>
  </div>
</div>
`;
